EPI/parse-nested-GrFN.py

```diff
@@ -0,0 +1,124 @@
+import json
+from argparse import ArgumentParser
+
+
+def formatGraph(data): 
+    modelMetadata = {}
+    nodesDict = {}
+    nodes = []
+    edges = []
+
+    for key,value in data.items():
+        if key == 'variables':
+            for item in value: 
+                if 'metadata' in item:
+                    metadata = item['metadata'][0]
+                else:
+                    metadata = {}
+                item['nodeType'] = 'variable'
+                item['metadata'] = metadata
+                nodesDict[item['uid']] = item
+        if key == 'functions':
+            for item in value: 
+                if 'metadata' in item:
+                    metadata = item['metadata'][0]
+                else:
+                    metadata = {} 
+                item['nodeType'] = 'function'
+                item['metadata'] = metadata
+                nodesDict[item['uid']] = item
+        if key == 'hyper_edges':
+            for item in value: 
+                if len(item['inputs']) > 0:
+                    for i in item['inputs']: # Inputs array is sometimes empty
+                        first_edge = { 'source': i, 'target': item['function']}
+                        edges.append(first_edge)
+                
+                for o in item['outputs']:
+                    second_edge = { 'source': item['function'], 'target': o }
+                    edges.append(second_edge)
+        if key == 'subgraphs':
+            for item in value: 
+                # Get parent name
+                parent_name = item['scope']
+                if (parent_name == '@global'):
+                    parent_name = 'root'
+
+                # Get container name
+                splitted_id = item['basename'].split('.')
+
+                #Get container metadata
+                if 'metadata' in item:
+                    metadata = item['metadata'][0]
```

Comment:
what if `'metadata'` is not in `item`, what value does `metadata` assume?

Reply:
I think it is just `{}` (empty)

Reply:
I believe it will take on the last value the `metadata` variable was assigned, above.

Reply:
Oh you're right. Thanks for the catch.